app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -333,16 +335,13 @@ class PipelineSampleReport extends React.Component {
 
       for (var i = 0; i < thresholded_taxons.length; i++) {
         let taxon = thresholded_taxons[i];
-
-        if (includedCategories.indexOf(taxon.category_name) >= 0) {
-          // In the included categories
-          selected_taxons.push(taxon);
-        } else if (
+        if (
+          includedCategories.indexOf(taxon.category_name) >= 0 ||
```

Comment:
perhaps we could deduplicate lines 339-342 and 357-360 by defining a helper function called `isIncluded(taxon, includedCategories, includedSubcategories)`, which should also help with readability. Particularly if you also move lines 330-334 in there